spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/features/TaskConfiguration.java

```diff
@@ -181,4 +194,17 @@ public TaskDefinitionRepository taskDefinitionRepository(DataSource dataSource)
 			return new RdbmsTaskDefinitionRepository(dataSource);
 		}
 	}
+
+	@EventListener
```

Comment:
Datastore access is not available at PostConstruct time. And is needed for registration.